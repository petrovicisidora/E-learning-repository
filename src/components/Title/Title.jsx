//Title component

import React from "react";

import { StyledTitle, StyledTitleContainer } from "./Title.style.js";

function Title({ title }) {
  return (
    <StyledTitleContainer>
      <StyledTitle>{title}</StyledTitle>
    </StyledTitleContainer>
  );
}

export default Title;
