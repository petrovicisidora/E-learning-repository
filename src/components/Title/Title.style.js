import styled from "styled-components";

export const StyledTitleContainer = styled.div`
  width: 100%;
  height: 20px;
  background-color: #c4c4c4a3;
  display: flex;
  align-items: center;
  margin-bottom: 20px;
  padding: 5px;
`;

export const StyledTitle = styled.span`
  font-weight: bolder;
  padding-left: 10px;
`;
