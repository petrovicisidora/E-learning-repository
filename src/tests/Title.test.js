import React from "react";
import { render, screen } from "@testing-library/react";

import Title from "../components/Title/Title";

test("renders title correctly", () => {
  const titleText = "Test title";
  render(<Title title={titleText} />);
  const titleElement = screen.getByText(titleText);

  expect(titleElement).toBeInTheDocument();
});

test("title has correct styles", () => {
  const titleText = "Test title";
  render(<Title title={titleText} />);
  const titleElement = screen.getByText(titleText);

  expect(titleElement).toHaveStyle("font-weight:bolder");
  expect(titleElement).toHaveStyle("padding-left:10px");
});

test("matches snapshot", () => {
  const { container } = render(<Title />);
  expect(container).toMatchSnapshot();
});
