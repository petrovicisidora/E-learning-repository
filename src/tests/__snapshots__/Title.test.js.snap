// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="sc-beySPh ktCImM"
  >
    <span
      class="sc-guDLey coRwsb"
    />
  </div>
</div>
`;
